Exclude disablemachExceptionHandler from header for tvOS and make sur…

TroubleMakerBen opened the pull request:
…e it’s not enabled when initializing MSCrashes on tvOS.

The API won't be available in the header, but I chose to leave the internal methods in there for tvOS.

jaelim-ms commented on MobileCenterCrashes/MobileCenterCrashesTests/MSCrashesTests.mm:
```diff
@@ -518,6 +518,14 @@ - (void)testDisableMachExceptionWorks {
 }
 #endif
 
+#if TARGET_OS_TV
```
You can change the order with previous test method and do 
`#if TARGET_OS_TV`
`#else`
`#endif`

jaelim-ms commented on MobileCenterCrashes/MobileCenterCrashes/Internals/MSCrashesPrivate.h:
```diff
@@ -70,10 +70,7 @@ typedef struct MSCrashesCallbacks {
 } MSCrashesCallbacks;
 
-// TODO: Mach exception handler is not supported on tvOS.
-#if !TARGET_OS_TV
```
I still don't know this is a right thing to have this property on tvOS. If it is not supported by tvOS, better to remove instead of keeping it. However if you are fine with change, I'm also okay.

clpolet replied:
I am fine as soon as it's not public.

clpolet commented on MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm:
```diff
@@ -229,13 +229,15 @@ + (MSErrorReport *_Nullable)lastSessionCrashReport {
+/*
+ * This can never be bound to Xamarin.
+ *
+ * This method is not part of the publicly available on tvOS as Mach exception handling is not possible on tvOS. 
```
"This method is not part of the publicly available **APIs** on tvOS" :)

TroubleMakerBen commented:
I just addressed the feedback. =)

![giphy-downsized](https://user-images.githubusercontent.com/10164200/29979434-4881e8bc-8efa-11e7-8a89-33556df97e54.gif)
